### ISSUE TITLE
Forward `IDisableCorsAttribute` and `IAntiforgeryPolicy` to the Abstractions package

### ISSUE THREAD

**Issue by dougbu** [aspnet/Mvc#7076]
`IDisableCorsAttribute` and `IAntiforgeryPolicy` (and `IgnoreAntiforgeryTokenAttribute`) are located in the Cors and ViewFeatures packages. This means external frameworks and extensions must reference those packages just to disable the possibly-conflicting features, even if they are aimed at API controllers. E.g. they may need to reference ViewFeatures to disable antiforgery in case their consumers reference that package -- but for no other reason.

Hit this while working on the ASP.NET Core WebHooks implementation. No action associated with the [`WebHookAttribute`](https://github.com/aspnet/WebHooks/blob/dev/src/Microsoft.AspNetCore.WebHooks.Receivers/WebHookAttribute.cs) needs antiforgery checks or CORS support. But, the containing Microsoft.AspNetCore.WebHooks.Receivers package would need to reference the Cors and ViewFeatures packages to disable both.

**Comment by dougbu** [comment]
Split out aspnet/CORS#134 and updated this description&hellip;

**Comment by mkArtakMSFT** [comment]
@dougbu, do you mean that we need to move a type to a different package? What does 'forward` mean?

**Comment by dougbu** [comment]
@mkArtakMSFT yes, I was talking about moving the interface and using type forwarding to avoid a breaking change.

**Comment by javiercn** [comment]
@dougbu
What configuration of the app would cause antiforgery to interfere with webhooks?
Are you trying to ensure that antiforgery is always disabled on those actions?

**Comment by dougbu** [comment]
> Are you trying to ensure that antiforgery is always disabled on those actions?

Yes.

WebHook actions are relatively normal MVC / Web API actions and some senders use form-URL encoded requests. Antiforgery tokens are not helpful in these machine-to-machine cases. So, this is about making it possible to disable antiforgery requirements without either mucking with the global configuration or bringing additional dependencies into the WebHook packages.

**Comment by rynowak** [comment]
Is this a problem that anyone has actually had?

**Comment by dougbu** [comment]
> Is this a problem that anyone has actually had?

It's an easily-avoided problem that users can hit with any of the (fairly simple) global antiforgery requirement configurations.

**Comment by javiercn** [comment]
@dougbu Can a customer do something in their application to avoid this?
* Put the `IgnoreAntiforgeryTokenAttribute` in their WebHooks applications. (Most people reference Microsoft.AspNetCore.App or Microsoft.AspNetCore.Mvc, so they have access to this).
* Write an application model convention that removes the AntiforgeryFilter from those controllers. (We can provide a sample for how to do this).

Or we can simply tell customers that they should not enable Antiforgery globally if they have WebHooks in their app/service.

**Comment by dougbu** [comment]
Customers can of course avoid this. Telling them avoidance is necessary isn't reliable.

Turning this around, why _not_ make it possible for WebHooks to detect this situation without taking numerous dependencies?
